packages/components/organisms/f-checkout/src/services/tests/analytics.test.js

```diff
@@ -0,0 +1,179 @@
+        });
+    });
+
+    describe('trackFormInteraction :: ', () => {
+        const action = 'start';
+        const error = 'postcodeNotCovered';
+
+        it('should call `event` method of `f-trak`', () => {
+            // Arrange
+            const expectedEvent = {
+                event: 'Form',
+                custom: {
+                    form: {
+                        name: 'checkout_guest',
+                        action,
+                        error,
+                        autofill: analyticsData.autofill,
+                        changes: analyticsData.changes
+                    }
+                }
+            };
+
+            // Act
+            trackFormInteraction(action, analyticsData, error);
+
+            // Assert
+            expect(trackEventSpy).toHaveBeenCalledWith(expectedEvent);
+        });
+
+        it.each([
+            ['checkout', true],
+            ['checkout_guest', false]
+        ])('should set `name` to %s if isLoggedIn is %s', (expectedName, isLoggedInState) => {
+            // Arrange
+            analyticsData.isLoggedIn = isLoggedInState;
+
+            const expectedEvent = {
+                event: 'Form',
+                custom: {
+                    form: {
+                        name: expectedName,
+                        action,
+                        error,
+                        autofill: analyticsData.autofill,
+                        changes: analyticsData.changes
+                    }
+                }
+            };
+
+            // Act
+            trackFormInteraction(action, analyticsData, error);
+
+            // Assert
+            expect(trackEventSpy).toHaveBeenCalledWith(expectedEvent);
+        });
+    });
+
+    describe('mapFieldNames :: ', () => {
+        it.each([
+            ['address.line1', 'addressLine1'],
+            ['line1', 'addressLine1'],
+            ['address.line2', 'addressLine2'],
+            ['line2', 'addressLine2'],
+            ['address.city', 'addressCity'],
+            ['city', 'addressCity'],
+            ['address.postcode', 'addressPostcode'],
+            ['postcode', 'addressPostcode'],
+            ['customer.firstName', 'firstName'],
+            ['customer.lastName', 'lastName'],
+            ['customer.mobileNumber', 'phone'],
+            ['mobileNumber', 'phone'],
+            ['customer.email', 'email']
+        ])('should correct %s to %s', (provided, expected) => {
+            // Act & Assert
+            expect(mapFieldNames(provided)).toEqual(expected);
+        });
+
+        it('should correctly map an array of field names', () => {
+            // Arrange
+            const provided = [
+                'mobilePhone',
+                'address.line1',
+                'customer.firstName',
+                'lastName',
+                'customer.email',
+                'city'
+            ];
+
+            const expected = [
+                'addressCity',
```

Comment:
More out of interest – does the function re-order the valued into alphabetical order here then? Or does the test just do a fuzzy match where the ordering doesn't matter when they're compared in the expect statement?

Reply:
It sorts alphabetically. I've added a test specifically for that now 👍